pkg/retriever/graphsyncretriever.go

```diff
@@ -91,9 +74,16 @@ type retrievalResult struct {
 // retrieval handles state on a per-retrieval (across multiple candidates) basis
 type graphsyncRetrieval struct {
 	*GraphSyncRetriever
-	ctx            context.Context
-	request        types.RetrievalRequest
-	eventsCallback func(types.RetrievalEvent)
+	ctx                 context.Context
+	request             types.RetrievalRequest
+	eventsCallback      func(types.RetrievalEvent)
+	candidateMetadata   map[peer.ID]*metadata.GraphsyncFilecoinV1
+	candidateMetdataMut *sync.RWMutex
```

Comment:
I don't think a pointer is necessary here, and it doesn't need to be initialised below in Retrieve(). Since it's only ever used by direct reference via its parent `graphsyncRetrieval` instance then by-value is fine.

Also, would you mind using the `Lk` suffix, we currently don't have any `*Mut` but we have plenty of `*Lk`:

```
$ git grep '[a-z]Mut.*sync' -- *.go | wc
      0       0       0
$ git grep '[a-z]Lk.*sync' -- *.go | wc
      8      27     593
```

Just decreases the cognitive overhead a tad.